jdaviz/configs/default/plugins/subset_tools/subset_tools.py

```diff
@@ -1119,8 +1124,22 @@ def _load_regions(self, regions, combination_mode=None, max_num_regions=None,
                 raise ValueError(f"{unknown_options} not one of {COMBO_OPTIONS}")
 
         previous_mode = self.app.session.edit_subset_mode.mode
+        previous_subset = self.app.session.edit_subset_mode.edit_subset
 
         with self.app._jdaviz_helper.batch_load():
```

Comment:
does `batch_load` accomplish anything here?  (I know it wasn't added in this PR - so may be out of scope to consider removing)

Reply:
It looks like `batch_load` delays linking the subsets until they are all created, which should speed up the operation quite a bit.